destination/destination_integration_test.go

```diff
@@ -0,0 +1,275 @@
+		sdk.StructuredData(want),
+	)
+
+	n, err := underTest.Write(ctx, []sdk.Record{rec})
+	is.NoErr(err)
+	is.Equal(1, n)
+
+	wID := uuid.NewMD5(uuid.NameSpaceOID, []byte(id)).String()
+	objects, err := client.Data().
+		ObjectsGetter().
+		WithClassName(cfg["class"]).
+		WithID(wID).
+		Do(ctx)
+	is.NoErr(err)
+	is.Equal(1, len(objects))
+
+	obj := objects[0]
+	got, ok := obj.Properties.(map[string]any)
+	is.True(ok) // expected object properties to be a map[string]any
+	is.Equal(want, got)
+}
+
+func TestDestination_Integration_Update(t *testing.T) {
+	integrationTest(t)
+
+	is := is.New(t)
+	openAIKey := os.Getenv("OPENAI_APIKEY")
+	is.True(openAIKey != "") // expected OPENAI_APIKEY to be set
+
+	ctx := context.Background()
+	class := fmt.Sprintf("Products_%v", time.Now().UnixMilli())
+	cfg := map[string]string{
+		"endpoint":           "localhost:18080",
+		"scheme":             "http",
+		"class":              class,
+		"moduleHeader.name":  "X-OpenAI-Api-Key",
+		"moduleHeader.value": openAIKey,
+		"generateUUID":       "true",
+	}
+
+	client, err := newWeaviateClient(cfg)
+	is.NoErr(err)
+	defer func() {
+		err = client.Schema().
+			ClassDeleter().
+			WithClassName(class).
+			Do(ctx)
+		is.NoErr(err)
+	}()
+
+	underTest := New()
+	err = underTest.Configure(ctx, cfg)
+	is.NoErr(err)
+
+	err = underTest.Open(ctx)
+	is.NoErr(err)
+
+	// Insert record
+	id := "test-id"
+	recInsert := sdk.Util.Source.NewRecordCreate(
+		sdk.Position("test-position"),
+		map[string]string{},
+		sdk.RawData(id),
+		sdk.StructuredData(map[string]any{
+			"product_name": "computer",
+			"price":        220.15,
+			"labels":       []any{"laptop", "navy-blue"},
+			"used":         true,
+		}),
+	)
+
+	n, err := underTest.Write(ctx, []sdk.Record{recInsert})
+	is.NoErr(err)
+	is.Equal(1, n)
+
+	// Update record
+	payloadUpdate := map[string]any{
+		"product_name": "computer",
+		"price":        330.75,
+		"labels":       []any{"laptop", "pink"},
+		"used":         true,
+	}
+	recUpdate := sdk.Util.Source.NewRecordUpdate(
+		sdk.Position("test-position"),
+		map[string]string{},
+		sdk.RawData(id),
+		nil,
+		sdk.StructuredData(payloadUpdate),
+	)
+
+	n, err = underTest.Write(ctx, []sdk.Record{recUpdate})
+	is.NoErr(err)
+	is.Equal(1, n)
+
+	// Verify update
+	wID := uuid.NewMD5(uuid.NameSpaceOID, []byte(id)).String()
+	objects, err := client.Data().
+		ObjectsGetter().
+		WithClassName(cfg["class"]).
+		WithID(wID).
+		Do(ctx)
+	is.NoErr(err)
+	is.Equal(1, len(objects))
+
+	obj := objects[0]
+	got, ok := obj.Properties.(map[string]any)
+	is.True(ok) // expected object properties to be a map[string]any
+	is.Equal(payloadUpdate, got)
+}
+
+func TestDestination_Integration_Delete(t *testing.T) {
+	integrationTest(t)
+
+	is := is.New(t)
+	openAIKey := os.Getenv("OPENAI_APIKEY")
+	is.True(openAIKey != "") // expected OPENAI_APIKEY to be set
+
+	ctx := context.Background()
+	class := fmt.Sprintf("products_%v", time.Now().UnixMilli())
+	cfg := map[string]string{
```

Comment:
this is the third test with the same configurations, so maybe it's worth returning the config from a function, and updating it if needed in the test?

Reply:
Yup, good suggestion!